src/state_machine.rs

```diff
@@ -322,6 +351,36 @@ impl StateMachine {
         }
     }
 
+    /// Query for a result, or yield: Err(NothingAvailable), Err(Disconnected).
+    #[cfg(feature = "use-mock-crust")]
+    pub fn try_step(&mut self, outbox: &mut EventBox) -> Result<(), TryRecvError> {
```

Comment:
This function just seems to be adding events in sequentiial order with timer ones always being at the end. I thought it was specifically decided not to do this, or has the requirement changed ?